src/solvers/cvc/cvc_conv.cpp

```diff
@@ -58,14 +58,552 @@ Function: cvc_convt::l_get
 
 tvt cvc_convt::l_get(literalt l) const
 {
-  if(l.is_true()) return tvt(true);
-  if(l.is_false()) return tvt(false);
+  if(l.is_true())
+    return tvt(true);
+  if(l.is_false())
+    return tvt(false);
   assert(l.var_no()<boolean_assignment.size());
   return tvt(boolean_assignment[l.var_no()]^l.sign());
 }
 
 /*******************************************************************\
 
+Function: cvc_convt::convert_binary_expr
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void cvc_convt::convert_binary_expr(const exprt &expr, const exprt &op)
+{
+  unsigned to_width=
+    unsafe_string2unsigned(id2string(expr.type().get(ID_width)));
+
+  if(op.type().id()==ID_signedbv)
+  {
+    unsigned from_width=
+      unsafe_string2unsigned(id2string(op.type().get(ID_width)));
+
+    if(from_width==to_width)
+      convert_expr(op);
+    else if(from_width<to_width)
+    {
+      out << "SX(";
+      convert_expr(op);
+      out << ", " << to_width << ")";
+    }
+    else
+    {
+      out << "(";
+      convert_expr(op);
+      out << ")[" << (to_width-1) << ":0]";
+    }
+  }
+  else if(op.type().id()==ID_unsignedbv)
+  {
+    unsigned from_width=
+      unsafe_string2unsigned(id2string(op.type().get(ID_width)));
+
+    if(from_width==to_width)
+      convert_expr(op);
+    else if(from_width<to_width)
+    {
+      out << "(0bin";
+
+      for(unsigned i=from_width; i<to_width; i++)
+        out << "0";
```

Comment:
A better way might be to use something like `setfill('0') << setw(to_width-from_width)`